src/Core/src/Platform/iOS/KeyboardAutoManagerScroll.cs

```diff
@@ -0,0 +1,614 @@
+			{
+				localCursor = tv.GetCaretRectForPosition(selectedRange.Start);
+				if (localCursor is CGRect local)
+					CursorRect = tv.ConvertRectToView(local, null);
+			}
+		}
+		else if (View is UITextField tf)
+		{
+			selectedTextRange = tf.SelectedTextRange;
+			if (selectedTextRange is UITextRange selectedRange)
+			{
+				localCursor = tf.GetCaretRectForPosition(selectedRange.Start);
+				if (localCursor is CGRect local)
+					CursorRect = tf.ConvertRectToView(local, null);
+			}
+		}
+
+		TextViewTopDistance = localCursor is CGRect cGRect ? 20 + (int)cGRect.Height : 20;
+
+		await AdjustPositionDebounce();
+	}
+
+	// Used to debounce calls from different oberservers so we can be sure
+	// all the fields are updated before calling AdjustPostition()
+	internal static async Task AdjustPositionDebounce()
+	{
+		DebounceCount++;
+		var entranceCount = DebounceCount;
+
+		await Task.Delay(10);
+
+		if (entranceCount == DebounceCount)
+		{
+			AdjustPosition();
+			DebounceCount = 0;
+		}
+	}
+
+	// main method to calculate and animate the scrolling
+	internal static void AdjustPosition()
+	{
+		if (View is not UITextField field && View is not UITextView)
+			return;
+
+		if (RootController is null)
+			return;
+
+		var rootViewOrigin = new CGPoint(RootController.Frame.GetMinX(), RootController.Frame.GetMinY());
+		var window = RootController.Window;
+
+		var kbSize = KeyboardFrame.Size;
+		var intersectRect = CGRect.Intersect(KeyboardFrame, window.Frame);
+		if (intersectRect == CGRect.Empty)
+			kbSize = new CGSize(KeyboardFrame.Width, 0);
+		else
+			kbSize = intersectRect.Size;
+
+		nfloat statusBarHeight;
+		nfloat navigationBarAreaHeight;
+
+		if (RootController.GetNavigationController() is UINavigationController navigationController)
+		{
+			navigationBarAreaHeight = navigationController.NavigationBar.Frame.GetMaxY();
+		}
+		else
+		{
+			if (OperatingSystem.IsIOSVersionAtLeast(13, 0))
+				statusBarHeight = window.WindowScene?.StatusBarManager?.StatusBarFrame.Height ?? 0;
+			else
+				statusBarHeight = UIApplication.SharedApplication.StatusBarFrame.Height;
+
+			navigationBarAreaHeight = statusBarHeight;
+		}
+
+		var topLayoutGuide = Math.Max(navigationBarAreaHeight, RootController.LayoutMargins.Bottom) + 5;
+
+		var keyboardYPosition = window.Frame.Height - kbSize.Height - TextViewTopDistance;
+
+		CGRect cursorRect;
+
+		if (CursorRect is CGRect cRect)
+			cursorRect = cRect;
+		else
+			return;
+
+		if (cursorRect.Y >= topLayoutGuide && cursorRect.Y < keyboardYPosition)
+			return;
+
+		nfloat move = 0;
+
+		// readjust contentInset when the textView height is too large for the screen
+		var rootSuperViewFrameInWindow = window.Frame;
+		if (RootController.Superview is UIView v)
+			rootSuperViewFrameInWindow = v.ConvertRectToView(v.Bounds, window);
+
+		if (cursorRect.Y > keyboardYPosition)
+			move = cursorRect.Y - keyboardYPosition;
+
+		else if (cursorRect.Y <= topLayoutGuide)
+			move = cursorRect.Y - (nfloat)topLayoutGuide;
```

Comment:
is this just an else?

Reply:
No because there is a scenario where the topLayoutGuide <  cursorRect.Y < keyboardYPosition and in this case, we don't want move to change value from 0